testSrc/unit/io/flutter/run/bazel/LaunchCommandsTest.java

```diff
@@ -230,32 +282,79 @@ public void producesCorrectCommandLineWithIosSimulator() throws ExecutionExcepti
   /**
    * Default configuration for the bazel test fields.
    */
-  private void setupBazelFields(BazelFields fields) {
-    fields.setEntryFile("/tmp/foo/lib/main.dart");
-    fields.setLaunchingScript("bazel-run.sh");
-    fields.setBazelTarget("bazel_target");
-    fields.setEnableReleaseMode(false);
+  private BazelFields setupBazelFields(
+    @Nullable String bazelTarget,
+    @Nullable String bazelArgs,
+    @Nullable String additionalArgs,
+    boolean enableReleaseMode) {
+    return new FakeBazelFields(new BazelFields(
+      "bazel_target",
+      bazelArgs,
+      additionalArgs,
+      enableReleaseMode
+    ));
+  }
+
+  private BazelFields setupBazelFields() {
+    return setupBazelFields("bazel_target", null, null, false);
   }
 
   /**
-   * Fake bazel test fields that doesn't depend on the Dart SDK.
+   * Fake bazel fields that doesn't depend on the Dart SDK.
    */
-  private static class FakeBazelTestFields extends BazelFields {
+  private static class FakeBazelFields extends BazelFields {
     MockVirtualFileSystem fs = new MockVirtualFileSystem();
+    final Workspace fakeWorkspace;
+
+    FakeBazelFields(@NotNull BazelFields template,
+                        @Nullable String daemonScript,
+                        @Nullable String doctorScript,
+                        @Nullable String launchScript,
+                        @Nullable String testScript) {
+      super(template);
+      fs.file("/workspace/WORKSPACE", "");
+      if (daemonScript != null) {
+        fs.file("/workspace/" + daemonScript, "");
+      }
+      if (doctorScript != null) {
+        fs.file("/workspace/" + doctorScript, "");
+      }
+      if (launchScript != null) {
+        fs.file("/workspace/" + launchScript, "");
+      }
+      if (testScript!= null) {
+        fs.file("/workspace/" + testScript, "");
+      }
+      fakeWorkspace = Workspace.forTest(
+        fs.findFileByPath("/workspace/"),
+        PluginConfig.forTest(
+          daemonScript,
+          doctorScript,
+          launchScript,
+          testScript
+        )
+      );
+    }
+
+    FakeBazelFields(@NotNull BazelFields template) {
+      this(
+        template,
+        "scripts/flutter-daemon.sh",
+        "scripts/flutter-doctor.sh",
+        "scripts/bazel-run.sh",
+        "scripts/flutter-test.sh"
+      );
+
+    }
+
 
     @Override
     void checkRunnable(@NotNull Project project) {}
 
-    @Override
     @Nullable
-    protected VirtualFile getAppDir(@NotNull Project project) {
-      if (getEntryFile() == null)
-        return null;
-      @NotNull
-      final String entryFile = getEntryFile();
-
-      // With the default entryFile from setupBazelFields, the application directory is the container of lib/main.dart.
-      return fs.refreshAndFindFileByPath(entryFile.replace("lib/main.dart", ""));
+    @Override
+    protected Workspace getWorkspace(@NotNull Project project) {
```

Comment:
This is that method-override-as-dependency-injection I mentioned above.  If we start adding more dependencies, this will get complicated to maintain.

For now, I think this is OK with just one dependency to mock/fake.